File: KakaoBlind2019/KakaoBlind2019-1.cpp
#include <iostream>
#include <vector>
#include <string>
#include <map>
using namespace std;

vector<string> solution(vector<string> record) {
    vector<string> answer;
    char a[50], b[50], c[50];
    map<string, string> mp;

    // Enter와 Change 마다 유저 아이디와 닉네임을 map으로 매핑 시켜줌
    for(int i=0; i<record.size(); i++) {
        sscanf(record[i].c_str(), "%s", a); // 문자열을 순차적으로 받아서 a에 넣는다

        if(a[0] == 'E') {
            sscanf(record[i].c_str(), "%s%s%s", a, b, c);
            mp[b] = c;
        }
        if(a[0] == 'C') {
            sscanf(record[i].c_str(), "%s%s%s", a, b, c);
            mp[b] = c;
        }
    }

    char output[50];
    for(int i=0; i<record.size(); i++) {
        sscanf(record[i].c_str(), "%s", a);

        if(a[0] == 'E') {
            sscanf(record[i].c_str(), "%s%s%s", a, b, c);
            sprintf(output, "%s님이 들어왔습니다.", mp[b].c_str());
            answer.push_back(output);
        }
        if(a[0] == 'L') {
            sscanf(record[i].c_str(), "%s%s", a, b);
            sprintf(output, "%s님이 나갔습니다.", mp[b].c_str());
            answer.push_back(output);
        }

    }
    return answer;
}

// int main() {
//     vector<string> record = {"Enter uid1234 Muzi", "Enter uid4567 Prodo","Leave uid1234","Enter uid1234 Prodo","Change uid4567 Ryan"};
//     vector<string> answer = solution(record);
//     for(int i=0; i<answer.size(); i++) cout << answer[i] << "\n";
// }

File: KakaoBlind2019/KakaoBlind2019-1.py
# 1. 오픈 채팅방

def solution(record):
    db_users = {}
    answer = []
    splited_record = []
    for item in record:
        splited_record = item.split(" ")

        if splited_record[0] == "Enter" or splited_record[0] == "Change":
            db_users[splited_record[1]] = splited_record[2]

    for item in record:
        splited_record = item.split(" ")

        if splited_record[0] == "Enter":
            answer.append(db_users[splited_record[1]] + "님이 들어왔습니다.")

        elif splited_record[0] == "Leave":
            answer.append(db_users[splited_record[1]] + "님이 나갔습니다.")

    return answer

input_record = ["Enter uid1234 Muzi","Enter uid4567 Prodo","Leave uid1234","Enter uid1234 Prodo","Change uid4567 Ryan"]
# 출력 예: ["Prodo님이 들어왔습니다.", "Ryan님이 들어왔습니다.", "Prodo님이 나갔습니다.", "Prodo님이 들어왔습니다."]

# input_record = ["Enter uid1434 Muzi", "Enter uid4567 Prodo", "Leave uid1434", "Enter uid1234 Prodo", "Change uid4567 Ryan", "Enter uid4067 AAA"]
# 출력 예: ["Muzi님이 들어왔습니다.", "Ryan님이 들어왔습니다.", "Muzi님이 나갔습니다.", "Prodo님이 나갔습니다.", "AAA님이 들어왔습니다."]
print("최종 결과: ", solution(input_record))
